pkg/kubectl/cmd/util/diff/diffoptions.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diff
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+)
+
+type DiffOptions struct {
+	FilenameOptions  resource.FilenameOptions
+	Mapper           meta.RESTMapper
+	Typer            runtime.ObjectTyper
+	Namespace        string
+	EnforceNamespace bool
+	DiffBuffer1      []byte
+	DiffBuffer2      []byte
+	Output           string
+	Codec            runtime.Encoder
+	Factory          cmdutil.Factory
+	Out              io.Writer
+	ErrOut           io.Writer
+	ext              string
+}
+
+func (o *DiffOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
+	if o.Output != "" {
```

Comment:
The `-o` flag is defaulted to `yaml`. Anything that is neither "yaml" nor "json" should return an error, including "".